resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -469,6 +474,12 @@ trait MesosSchedulerUtils extends Logging {
         .setType(Value.Type.RANGES)
         .setRanges(Value.Ranges.newBuilder().addRange(rangeValue))
       role.foreach(r => builder.setRole(r))
+
+      if (role.forall(r => !r.equals("*"))) {
```

Comment:
Maybe abstract this logic into a general function?

Reply:
ok sure.

Reply:
Oh, I thought you could put the whole block into a function, that way if we need to change the way we do the protos some day it's easy to change in one place.